### PR TITLE
Add issuer option to type/provider

### PR DESCRIPTION
**Pull request by olliewalsh**
Adds an optional issuer, which can be used to target a FreeIPA sub-CA.

### CODE REVIEW

**Comment by olliewalsh** [comment]
@JAORMX could you take a look?

**Review by JAORMX — APPROVED** [review]

File: lib/puppet/provider/certmonger_certificate/certmonger_certificate.rb [hunk 1]
```diff
@@ -75,6 +75,16 @@ def self.parse_cert_list(list_output)
...
             current_cert[:hostname] = cn_match[1]
           end
+        when %r{^\s+issuer: .*}
+          # FIXME(owalsh): This is also hacky! Use an actual library to parse
+          # the issuer DN.
+          subj_match = line.match(%r{issuer: (.*)})
+          if subj_match[1].empty?
+            current_cert[:issuer] = ''
+          else
+            cn_match = line.match(%r{issuer: .*CN=(.*?)(?:,.*|$)})
```
Comment (JAORMX) [inline lib/puppet/provider/certmonger_certificate/certmonger_certificate.rb:85]: Uhm... What would this match?

Comment (olliewalsh) [reply lib/puppet/provider/certmonger_certificate/certmonger_certificate.rb:85]: Bah, I mistakenly thought that the issuer CN always matched the issuer name. Didn't realised that I just happened to create the sub-CA this way in FreeIPA

Comment (JAORMX) [reply lib/puppet/provider/certmonger_certificate/certmonger_certificate.rb:85]: That seems to be the case. I suggest you just match the whole issuer string.

Comment (olliewalsh) [reply lib/puppet/provider/certmonger_certificate/certmonger_certificate.rb:85]: nope, that won't work either. certmonger accepts the issuer name when creating the request but reports the issuer_dn when querying.

Looks like the defined type hit this too, and requires both the issuer name and issuer DN: https://github.com/saltedsignal/puppet-certmonger/blob/master/manifests/request_ipa_cert.pp#L133

**Comment by olliewalsh** [comment]
certmonger support for freeipa sub-CAs has other problems:
-it doesn't manage the sub-CA cert (e.g doesn't renew it etc...)
-saving the sub-CA cert doesn't work (if -F is used then the root CA cert is saved, not the sub-CA cert)

Abandoning this change for now until we have a better solution for sub-CAs

**Pull request closed without merging**